docs/source/nlp/joint_intent_slot.rst

```diff
@@ -3,10 +3,253 @@
-`NeMo/tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb <https://github.com/NVIDIA/NeMo/blob/main/tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb>`__.
+Joint Intent and Slot classification is a method for classifying an Intent and detecting all
+relevant Slots (Entities) for the Intent in a query.
+For example, in the query `What is the weather in Santa Clara tomorrow morning?`,
+we would like to classify the query as a `weather intent`, detect `Santa Clara` as a `location slot`,
+and `tomorrow morning` as a `date_time slot`. Intent and Slot names are usually task-specific and
+defined as labels in the training data. This is a fundamental step that is executed in any
+task-driven conversational assistant.
 
-Connect to an instance with a GPU (Runtime -> Change runtime type -> select "GPU" for hardware accelerator)
+Our BERT-based model implementation allows you to train and detect both of these tasks together.
 
-An example script on how to train the model could be found here: `NeMo/examples/nlp/intent_slot_classification <https://github.com/NVIDIA/NeMo/tree/main/examples/nlp/intent_slot_classification>`__.
+
+.. note::
+
+    We recommend you try this model in a Jupyter notebook \
+    (can run on `Google's Colab <https://colab.research.google.com/notebooks/intro.ipynb>`_.): \
+    `NeMo/tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb <https://github.com/NVIDIA/NeMo/blob/main/tutorials/nlp/Joint_Intent_and_Slot_Classification.ipynb>`__.
+
+    Connect to an instance with a GPU (Runtime -> Change runtime type -> select "GPU" for hardware accelerator)
+
+    An example script on how to train the model could be found here: `NeMo/examples/nlp/intent_slot_classification <https://github.com/NVIDIA/NeMo/tree/main/examples/nlp/intent_slot_classification>`__.
+
+
+Data Format
+-----------
+
+When training the model, the dataset should be first converted to the required data format,
+which requires these files:
+
+- :code:`dict.intents.csv` - A list of all Intent names in the data. Only one line per intent name
+  is allowed.
+- :code:`dict.slots.csv` - A list of all Slot names in the data. Only one line per intent name
+  is allowed.
+- :code:`train.tsv/test.tsv` - A list of original queries, one per line, with the intent number
+  separated by a tab (e.g. "what alarms do i have set right now <TAB> 0"). Intent numbers are
+  set according to the intent line in the intent dictionary file (:code:`dict.intents.csv`),
+  starting from 0. The first line in these files should contain the header line "sentence
+  <tab> label".
+- :code:`train_slot.tvs/test_slot.tsv` - A list that contains one line per a query, when an each word from the original text queries,
+  is replaced by a token number from the slots dictionary file (dict.slots.csv), counted starting from 0.
+  All the words which do not contain a relevant slot are replaced by 'out-of scope' token number, which is also a part of the slot dictionary file,
```

Comment:
Could you provide an example of dicts files? this part `'out-of scope' token number, which is also a part of the slot dictionary file,` is unclear without an example

Reply:
Done

Reply:
Done